README.md

```diff
@@ -55,7 +65,197 @@ asyncio.run(main())
+
+- **Payload generation**: Decorators add a `payload()` method to generate execution payloads
+- **Session management**: You manually create sessions and control when to register modules
+- **Function grouping**: Group related functions into modules for efficient reuse
+- **Response handling**: Choose between `session.execute()` for full response (status, logs, return value) or `session.rpc()` for just the return value
+
+The Functional API offers two decorators: `@d3pythonscript` and `@d3function`:
+- **`@d3pythonscript`**: 
+  - Does not require registration.
+  - Best for simple scripts executed once or infrequently.
+- **`@d3function`**:
+  - Must be registered on Designer before execution.
+  - Functions decorated with the same `module_name` are grouped together and can call each other, enabling function chaining and code reuse.
+  - Registration is automatic when you pass module names to the session context manager (e.g., `D3AsyncSession('localhost', 80, ["mymodule"])`). If you don't provide module names, no registration occurs.
+
+### Session API Methods
+
+Both `D3AsyncSession` and `D3Session` provide two methods for executing functions:
+
+- **`session.rpc(payload)`** - Returns only the return value from the function execution. Simpler for most use cases.
+- **`session.execute(payload)`** - Returns a `PluginResponse` object containing:
+  - `returnValue`: The function's return value
+  - `status`: Execution status (code, message, details)
+  - `d3Log`: Designer console output during execution
+  - `pythonLog`: Python-specific output (print statements, warnings)
+
+**Example**
+
+```python
+from designer_plugin.d3sdk import d3pythonscript, d3function, D3AsyncSession
+from typing import TYPE_CHECKING
+if TYPE_CHECKING:
+    from designer_plugin.pystub.d3 import *
+
+# 1. @d3pythonscript - simple one-off execution
+@d3pythonscript
+def rename_surface(surface_name: str, new_name: str):
+    surface: Screen2 = resourceManager.load(
+        Path(f'objects/screen2/{surface_name}.apx'),
+        Screen2)
+    surface.rename(surface.path.replaceFilename(new_name))
+
+# 2. @d3function - reusable module-based functions
+@d3function("mymodule")
+def rename_surface_get_time(surface_name: str, new_name: str) -> str:
+    surface: Screen2 = resourceManager.load(
+        Path(f'objects/screen2/{surface_name}.apx'),
+        Screen2)
+    surface.rename(surface.path.replaceFilename(new_name))
+    return my_time()  # Call other functions in the same module
+
+@d3function("mymodule")
+def my_time() -> str:
+    import datetime
+    return str(datetime.datetime.now())
+
+# Usage with async session
+async with D3AsyncSession('localhost', 80, ["mymodule"]) as session:
+    # d3pythonscript: no registration needed
+    await session.rpc(rename_surface.payload("surface 1", "surface 2"))
```

Comment:
At one point we had an interface something like:
`result = function.execute(args)`
which was quite simple and intuitive. Did we decide against this approach in the end?